docs/classes/server.md

```diff
@@ -3,71 +3,103 @@ title: Server
 permalink: /classes/server
 ---
 
-{:.warning}
-> This page needs an update.
-
 * Table of Contents
 {:toc}
 
 The `Server` instance controls the whole listening and dispatches the parsed requests.
 
-## `attach(ServerObserver)`
+## Constructor
+
+```php
+public function __construct(
+    Amp\Socket\Server[] $servers,
+    RequestHandler $requestHandler,
+    Psr\Log\LoggerInterface $logger,
+    Options $options = null
+)
+```
+
+### Parameters
 
-Enables a [`ServerObserver`](serverobserver.md) instance to be notified of the updates.
+|[`Amp\Socket\Server[]`](https://amphp.org/socket/server)|`$servers`|List of socket servers.|
+|[`RequestHandler`](request-handler.md)|`$requestHandler`|Request handler interface.|
+|[`Psr\Log\LoggerInterface`]()|`$logger`|A PSR compliant logger (eg. [amphp/log](https://github.com/amphp/log)).|
+|[`Options`](options.md)<br />`null`|`$options`|HTTP server settings.<br />`null` creates an `Options` object with all default options.|
 
-## `detach(ServerObserver)`
+{:.warning}
+> `$servers` must be a non-empty list of [`Amp\Socket\Server`](https://amphp.org/socket/server) objects. Otherwise an [`\Error`](http://php.net/manual/en/class.error.php) will be thrown.
+
+## `setDriverFactory(HttpDriverFactory $driverFactory)`
+
+Define a custom HTTP driver factory.
 
-Disables notifications for the passed [`ServerObserver`](serverobserver.md) instance.
+## `setErrorHandler(ErrorHandler $errorHandler)`
 
-## `state()`
+Sets the error handler instance to be used for generating error responses.
 
-Gets the current server state, which is one of the following class constants:
+## `getState(): int`
+
+Returns the current server state, which is one of the following class constants:
 
 * `Server::STARTING`
 * `Server::STARTED`
 * `Server::STOPPING`
 * `Server::STOPPED`
 
-## `getOption(string)`
+## `getOptions(): Options`
+
+Returns the server options object.
+
+## `getErrorHandler(): ErrorHandler`
 
-Gets an [`option`](options.md) value.
+Returns the error handler.
 
-## `setOption(string, $value)`
+## `getLogger(): Psr\Log\LoggerInterface`
 
-Sets an [`option`](options.md) value.
+Returns the logger.
 
-## `stop(): Promise`
+## `getTimeReference(): TimeReference`
 
-Initiate shutdown sequence. The returned `Promise` will resolve when the server has successfully been stopped.
+Returns the time context.
 
-## `monitor(): array`
+## `attach(ServerObserver $observer)`
 
-See [`Monitor`](monitor.md), it returns an array of the following structure:
+Enables a [`ServerObserver`](server-observer.md) instance to be notified of the updates.
+
+## `start(): Promise`
+
+Starts the server.
+
+## `stop(int $timeout = 3000ms): Promise`
+
+Stops the server. `$timeout` is the number of milliseconds to allow clients to gracefully shutdown before forcefully closing.
+
+The returned `Promise` will resolve when the server has successfully been stopped.
+
+## `__debugInfo(): array`
```

Comment:
IMO `__debugInfo()` shouldn't be documented or explicitly documented as not being public API.